docs/tutorials/custom_experiment.rst

```diff
@@ -0,0 +1,559 @@
+    """Randomized measurement experiment."""
+        def __init__(
+            self,
+            circuit,
+            measured_qubits=None,
+            physical_qubits=None,
+            backend=None,
+            **experiment_options
+        ):
+            """Basic randomized Z-basis measurement experiment via a Pauli frame transformation
+            
+            Note this will just append a new set of measurements at the end of a circuit.
+            A more advanced version of this experiment would be to use a transpiler pass to
+            replace all existing measurements in a circuit with randomized measurements.
+            """
+            if physical_qubits is None:
+                physical_qubits = tuple(range(circuit.num_qubits))
+            if measured_qubits is None:
+                measured_qubits = tuple(range(circuit.num_qubits))
+            
+            # Initialize BaseExperiment
+            analysis = RandomizedMeasurementAnalysis()
+            super().__init__(physical_qubits, analysis=analysis, backend=backend)
+            
+            # Add experiment properties
+            self._circuit = circuit        
+            self._measured_qubits = measured_qubits
+            
+            # Set any init optinos
+            self.set_experiment_options(**experiment_options)
+
+Notice that when we called ``super().__init__``, we provided the list of physical qubits,
+the name of our analysis class, and the backend, which is optionally specified by the
+user at this stage.
+
+
+Now we consider default experiment options. Because randomness is involved,
+it is good practice to allow the user to set a seed. We would also like the user to 
+be able to set how many repetitions of the circuit to run:
+
+.. jupyter-input::
+
+    ...
+
+        @classmethod
+        def _default_experiment_options(cls):
+            options = super()._default_experiment_options()
+            options.num_samples = None
+            options.seed = None
+            return options
+
+
+Now we write the ``circuits()`` method. We need to take the input circuit in
+``self._circuit`` and add our random Paulis as well as measurement at the end. We use
+the built-in property :attr:`~.BaseExperiment.num_qubits` of :class:`~.BaseExperiment`
+to get the number of qubits in the experiment. We keep track of the list of qubits and
+classical Note that the circuits themselves are always built on qubits `0` to `length of
+the circuit - 1`, and not the actual physical qubit indices given in
+``physical_qubits``, as discussed in :doc:`getting_started`.
+
+.. jupyter-input::
+
+    ...
+
+
+        def circuits(self):
+            # Number of classical bits of the original circuit
+            circ_nc = self._circuit.num_clbits
+
+            # Number of added measurements
+            meas_nc = len(self._measured_qubits)
+
+            # Classical bits of the circuit
+            circ_clbits = list(range(circ_nc))
```

Comment:
Maybe somewhere you want to mention that the input circuit could mid-circuit measurements? Or maybe for the sake of the example it is cleaner to assume no extra measurements and the measurements added by the experiment are the only measurements?

Reply:
I added a simple mid-circuit measurement circuit at the end to demonstrate this, though I don't run it. Maybe there's a more interesting example to actually run?